PostHog/PostHogSDK.swift

```diff
@@ -151,6 +159,10 @@ let maxRetryDelay = 30.0
             properties["$groups"] = groups!
         }
 
+        if let sessionId = sessionId {
+            properties["$session_id"] = sessionId
+        }
```

Comment:
Those checks have to be locked as well, because between the `sessionId` check and assigning from `sessionId`, the value might have changed already, the chance is almost nothing but it's possible so better to lock to avoid racy conditions that would be very hard to debug.